### PR TITLE
ci(satp-hermes): add satp-hermes to ci

### LINKED ISSUE #79: Add SATP tests to CI

### PR DESCRIPTION
**Pull request by AndreAugusto11**
* Add SATP-Hermes to CI
* Fix some tests that were hanging the terminal after execution

closes #79

### CODE REVIEW

**Comment by AndreAugusto11** [comment]
Note: it seems like the ci only runs successfully with the current regex `packages/cactus-plugin-satp-hermes/src/test/typescript/(unit/.*/*.test.ts|integration/(client-.*.test.ts|odap.test.ts|odap-api.*.test.ts|server-.*.test.ts|backup-.*.test.ts|odap-rollback-.*.test.ts))`, and not with the most general (and usual) pattern for all packages. Since we are going to refactor the implementation, let's just keep it like this. It is enough for us, just to make sure we are not breaking anything.

**Review by petermetz — APPROVED** [review]
LGTM

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1728,33 +1728,31 @@ jobs:
...
-  #     - run: ./tools/ci.sh
+  cactus-plugin-satp-hermes:
+    continue-on-error: false
+    env:
+      FULL_BUILD_DISABLED: true
+      JEST_TEST_RUNNER_DISABLED: false
+      JEST_TEST_PATTERN: packages/cactus-plugin-satp-hermes/src/test/typescript/(unit/.*/*.test.ts|integration/(client-.*.test.ts|odap.test.ts|odap-api.*.test.ts|server-.*.test.ts|backup-.*.test.ts|odap-rollback-.*.test.ts))
+      TAPE_TEST_RUNNER_DISABLED: true
+    needs: build-dev
+    runs-on: ubuntu-20.04
```
Comment (petermetz) [inline .github/workflows/ci.yaml:1739]: @AndreAugusto11 Sorry, one more thing: please upgrade to 22.04 (we have a pr on the upstream repo that mass upgrades everything to 22.04)

Comment (AndreAugusto11) [reply .github/workflows/ci.yaml]: Thanks @petermetz!

**Comment by petermetz** [comment]
@AndreAugusto11 While the ubuntu-22.04 upgrade could technically cause issues, it is very very unlikely that it did. The timeouts are most likely because the tests are flaky to begin with due to the flakiness of the Fabric AIO container image prior to my most recent fixes that migrated it to use the current LTS v2.5.6 Fabric under the hood. 

I rebased this to upstream/main and then applied the migration steps to the test cases and they are all passing fine (before that all the tests were broken because of the old Fabric 2 AIO image they use)

[2ca4d48fa405bf69bbf8182af4d4ffb68d631e2d](https://github.com/RafaelAPB/blockchain-integration-framework/commit/3421ebc1f020545b11e3146d8569937ddcc6687d)

I don't know how were they passing earlier, but it was must likely just blind luck with the flaky tests.

What I'd do is rebase the base branch (Rafael's dev branch) onto upstream main and then rebase this onto that and then apply/cherry pick my commit linked above and that should have the tests passing regardless of ubuntu 20 or 22.

**Comment by AndreAugusto11** [comment]
@petermetz thank you for the patch! Can I squash both commits now and open a PR in the main repo? Would you like me to change your commit message to be by the commit rules?

**Comment by petermetz** [comment]
> @petermetz thank you for the patch! It's fixed!! Can I squash both commits now and open a PR in the main repo? Would you like me to change your commit message to be by the commit rules?

@AndreAugusto11 It's OK to squash, yes. The `!squash` prefix in my otherwise invalid looking commit subject (the first line) is a convention of the _conventional commits_ standard that means something along the lines of "this commit is not meant to be merged on it's own but only as part of another commit and therefore does not contain a valid commit subject line to explicitly signal that" in short, if you see any commit of mine in the future that's also starting with `!squash` then you know it's meant for squashing and no need to even ask. :-)

**Comment by AndreAugusto11** [comment]
> > @petermetz thank you for the patch! It's fixed!! Can I squash both commits now and open a PR in the main repo? Would you like me to change your commit message to be by the commit rules?
> 
> @AndreAugusto11 It's OK to squash, yes. The `!squash` prefix in my otherwise invalid looking commit subject (the first line) is a convention of the _conventional commits_ standard that means something along the lines of "this commit is not meant to be merged on it's own but only as part of another commit and therefore does not contain a valid commit subject line to explicitly signal that" in short, if you see any commit of mine in the future that's also starting with `!squash` then you know it's meant for squashing and no need to even ask. :-)

Thank you for the explanation @petermetz!!

**Pull request merged**